sale_financial_risk_pos_compatibility/static/src/js/CreateOrderPopup.esm.js

```diff
@@ -0,0 +1,89 @@
+/** @odoo-module **/
+
+import CreateOrderPopup from "point_of_sale.CreateOrderPopup";
+import Registries from "point_of_sale.Registries";
+import framework from "web.framework";
+
+export const CreateOrderPopupRisk = (CreateOrderPopup) =>
+    class CreateOrderPopupRisk extends CreateOrderPopup {
+        async _actionCreateSaleOrder(order_state) {
+            if (order_state === "draft") {
+                // Default behavior
+                return await super._actionCreateSaleOrder(order_state);
+            }
+            this.extraContext = false;
+            const order = this.env.pos.get_order();
+            const risk_amount =
+                order.get_total_with_tax() + order.get_rounding_applied();
+            const [partner] = await this.rpc({
+                model: "res.partner",
+                method: "read",
+                args: [
+                    order.partner.id,
+                    [
+                        "risk_exception",
+                        "risk_sale_order_limit",
+                        "risk_sale_order",
+                        "risk_sale_order_include",
+                        "risk_total",
+                        "credit_limit",
+                    ],
+                ],
+            });
+            let exception_msg = "";
+            if (partner.risk_exception) {
+                exception_msg = this.env._t("Financial risk exceeded.\n");
+            } else if (
+                partner.risk_sale_order_limit &&
+                partner.risk_sale_order + risk_amount > partner.risk_sale_order_limit
+            ) {
+                exception_msg = this.env._t(
+                    "This sale order exceeds the sales orders risk.\n"
+                );
+            } else if (
+                partner.risk_sale_order_include &&
+                partner.risk_total + risk_amount > partner.credit_limit
+            ) {
+                exception_msg = this.env._t(
+                    "This sale order exceeds the financial risk.\n"
+                );
+            } else {
+                return await super._actionCreateSaleOrder(order_state);
+            }
+            const {confirmed} = await this.showPopup("ConfirmRiskPopup", {
+                title: "Partner risk exceeded",
+                body: exception_msg,
+                showButton: this.env.pos.user.has_role_risk_manager,
+            });
+            if (confirmed) {
+                this.extraContext = {context: {bypass_risk: true}};
```

Comment:
Instead of using this variable, I suggest setting this value as an order property;

```js
order.set_bypass_risk(true);
```
and order model:

```js
const PosSaleFinancialRiskOrder = (Order) =>
    class PosSaleFinancialRiskOrder extends Order {
        constructor() {
            super(...arguments);
            this.bypass_risk = false;
        }
        async set_bypass_risk(bypass_risk) {
            this.bypass_risk = bypass_risk
        }
        export_as_JSON() {
            const result = super.export_as_JSON(...arguments);
            result.bypass_risk = this.bypass_risk;
            return result;
        }
        init_from_JSON(json) {
            super.init_from_JSON(...arguments);
            this.bypass_risk = json.bypass_risk;
        }
    }
    
Registries.Model.extend(Order, PosSaleFinancialRiskOrder);

export default PosSaleFinancialRiskOrder;
```

Reply:
so this values will available in `order_data` of `def create_order_from_pos(self, order_data, action)` method

and you can check:
```python
if "bypass_risk" in order_data:
    self = self.with_context(bypass_risk=bypass_risk)
return super(SaleOrder, self).create_order_from_pos(order_data, action)
```

please check this way